doc/status-current-code.md

```diff
@@ -0,0 +1,63 @@
+## Current Network Code Overview
+
+The main responsabilities of Network configuration resides between yast2-network 
+and yast2 (NetworkInterfaces) and them are by sure the more complex parts of
+the code.
+
+- NetworkInterfaces (yast2): represent mainly the configured interfaces and is in
+  charge of writing the ifcfg-giles. The numbers of bugs in this area are not
+  so big, we have been dealing mostly with some normalization of the config or
+  better defaults like for wireless configuration.
+
+- LanItems (yast2-network): It was introduced in 2007 [1] for unifying all the
+  maps in a single list.
+
+  So basically each LanItems Item has the information about its (hwinfo, ifcfg
+  file, [udev rules](udev-rules-implementation.org), and table description 
+  (the overview))
+
+  The match between these different objects / hashes is based on the interface 
+  or device name, and as we can see it has been one of me most common point of 
+  failures in network configuration.
```

Comment:
can you please add links to some well known bugs in network so we know which use cases we need to have in mind? Also one issue which is not mentioned here is problem that not all hwinfo need to have ifcfg file and so one, which sometimes causes problem.

Reply:
Sure, that is the main difference between virtual interfaces and phisical ones, and already described in Michal Zugec email link. So, will try to expand here in a way that the problems behind that are easy to expose and understand.

I will add also a section with the installation overview (not having a separation between the inst-sys network configuration and the proposal for the system being installed).

Reply:
moreover Items use a useless layer of abstraction (item_id). Designing a structure where devices would be more easily accessible according its names would have bigger benefit (now we search such device with O(n) complexity)